From: Luca Weiss <luca.weiss@fairphone.com>
Date: Mon, 8 May 2023 09:34:54 +0200
Subject: Revert "arm64: dts: qcom: sm7225-fairphone-fp4: WIP Force camera
 clock & regs on"

This reverts commit 531e04a5fca976447c5b49c665baff7ddd1fb48e.
---
 arch/arm64/boot/dts/qcom/sm7225-fairphone-fp4.dts | 55 -----------------------
 drivers/i2c/busses/i2c-qcom-cci.c                 |  1 -
 2 files changed, 56 deletions(-)

diff --git a/arch/arm64/boot/dts/qcom/sm7225-fairphone-fp4.dts b/arch/arm64/boot/dts/qcom/sm7225-fairphone-fp4.dts
index 2c2c59e..f9f397c 100644
--- a/arch/arm64/boot/dts/qcom/sm7225-fairphone-fp4.dts
+++ b/arch/arm64/boot/dts/qcom/sm7225-fairphone-fp4.dts
@@ -113,33 +113,6 @@ active-config0 {
 			};
 		};
 	};
-
-	fake-reg-mclk0 {
-		compatible = "regulator-fixed-clock";
-		regulator-name = "fake-reg-mclk0";
-		pinctrl-names = "default";
-		pinctrl-0 = <&cam_mclk0_default>;
-		clocks = <&camcc CAM_CC_MCLK0_CLK>;
-		regulator-always-on;
-	};
-
-	fake-reg-mclk1 {
-		compatible = "regulator-fixed-clock";
-		regulator-name = "fake-reg-mclk1";
-		pinctrl-names = "default";
-		pinctrl-0 = <&cam_mclk1_default>;
-		clocks = <&camcc CAM_CC_MCLK1_CLK>;
-		regulator-always-on;
-	};
-
-	fake-reg-mclk2 {
-		compatible = "regulator-fixed-clock";
-		regulator-name = "fake-reg-mclk2";
-		pinctrl-names = "default";
-		pinctrl-0 = <&cam_mclk2_default>;
-		clocks = <&camcc CAM_CC_MCLK2_CLK>;
-		regulator-always-on;
-	};
 };
 
 &adsp {
@@ -586,49 +559,42 @@ &pm8008_l1 {
 	regulator-min-microvolt = <528000>;
 	regulator-max-microvolt = <1200000>;
 	regulator-min-dropout-voltage-microvolt = <110000>;
-	regulator-always-on; // FIXME ultra-wide cam
 };
 
 &pm8008_l2 {
 	regulator-min-microvolt = <528000>;
 	regulator-max-microvolt = <1200000>;
 	regulator-min-dropout-voltage-microvolt = <110000>;
-	regulator-always-on; // FIXME main cam DVDD
 };
 
 &pm8008_l3 {
 	regulator-min-microvolt = <1500000>;
 	regulator-max-microvolt = <2800000>;
 	regulator-min-dropout-voltage-microvolt = <128000>;
-	regulator-always-on; // FIXME front cam
 };
 
 &pm8008_l4 {
 	regulator-min-microvolt = <1500000>;
 	regulator-max-microvolt = <2904000>;
 	regulator-min-dropout-voltage-microvolt = <96000>;
-	regulator-always-on; // FIXME ultra-wide cam
 };
 
 &pm8008_l5 {
 	regulator-min-microvolt = <1500000>;
 	regulator-max-microvolt = <2900000>;
 	regulator-min-dropout-voltage-microvolt = <128000>;
-	regulator-always-on; // FIXME main cam AVDD
 };
 
 &pm8008_l6 {
 	regulator-min-microvolt = <1500000>;
 	regulator-max-microvolt = <1800000>;
 	regulator-min-dropout-voltage-microvolt = <56000>;
-	regulator-always-on; // FIXME CCI bus pull-up & ultra-wide cam & main cam DOVDD & front cam
 };
 
 &pm8008_l7 {
 	regulator-min-microvolt = <1500000>;
 	regulator-max-microvolt = <3140000>;
 	regulator-min-dropout-voltage-microvolt = <120000>;
-	regulator-always-on; // FIXME main cam OIS power
 };
 
 &pmk8350_adc_tm {
@@ -773,27 +739,6 @@ qup_uart1_sleep_tx: qup-uart1-sleep-tx-state {
 		bias-pull-up;
 	};
 
-	cam_mclk0_default: cam-mclk0-default {
-		pins = "gpio29";
-		function = "cam_mclk0";
-		drive-strength = <2>;
-		bias-disable;
-	};
-
-	cam_mclk1_default: cam-mclk1-default {
-		pins = "gpio30";
-		function = "cam_mclk1";
-		drive-strength = <2>;
-		bias-disable;
-	};
-
-	cam_mclk2_default: cam-mclk2-default {
-		pins = "gpio31";
-		function = "cam_mclk2";
-		drive-strength = <2>;
-		bias-disable;
-	};
-
 	pm8008_active: pm8008-active-state {
 		pins = "gpio58";
 		function = "gpio";
diff --git a/drivers/i2c/busses/i2c-qcom-cci.c b/drivers/i2c/busses/i2c-qcom-cci.c
index 22ec7cd..414882c 100644
--- a/drivers/i2c/busses/i2c-qcom-cci.c
+++ b/drivers/i2c/busses/i2c-qcom-cci.c
@@ -1,7 +1,6 @@
 // SPDX-License-Identifier: GPL-2.0
 // Copyright (c) 2012-2016, The Linux Foundation. All rights reserved.
 // Copyright (c) 2017-2022 Linaro Limited.
-#define DEBUG
 
 #include <linux/clk.h>
 #include <linux/completion.h>
